### PR TITLE
[chore]: minimize eslint setting overrides

### LINKED ISSUE #1258: [chore]: minimize eslint setting overrides
### Requirement

To keep the code in the best forward-compatible shape with linters, it's best to minimize the number of lint rule overrides.

### Problem

https://github.com/jaegertracing/jaeger-ui/pull/1256#discussion_r1133295150

### Proposal

make the root `eslintrc` also extend the recommended config from `@typescript-eslint`, because that config already does this toggling for us. This is what `plexus` does already. 

### Open questions

_No response_

### PR DESCRIPTION
**Pull request by anikdhabal**
## Which problem is this PR solving?
Resolves #1258 

## Short description of the changes
made the root eslintrc also extend the recommended config from @typescript-eslint.

### CODE REVIEW

**Comment by anikdhabal** [comment]
> What could be the testing methodology for this PR? I.e., just having a green CI seems insufficient as it doesn't prove that a group of files are not getting excluded from linting after the change.

I've tested it locally using the "eslint", but I don't think it is relevant. Any suggestion?

**Comment by yurishkuro** [comment]
maybe introducing some errors in the files and observing that linting errors are reported

**Comment by anikdhabal** [comment]
> maybe introducing some errors in the files and observing that linting errors are reported

Yes, I've debugging and it is inducing "linting issue" in many files. Due to typescript-eslint/recommended.

**Comment by anikdhabal** [comment]
> maybe introducing some errors in the files and observing that linting errors are reported

@yurishkuro Instead of extend the recommended config from @typescript-eslint if we only minimize the setting overrides and also make root eslintrc , then it would be fine. Due to @typescript-eslint/recomended several linting has been occurred in many files.

**Comment by anikdhabal** [comment]
@yurishkuro can you review it once?

File: .eslintrc.js [hunk 1]
```diff
@@ -95,6 +77,12 @@ module.exports = {
     'no-self-compare': 0,
     'no-underscore-dangle': 0,
     'prefer-destructuring': 0,
+    
+    /* tsx */
+    "@typescript-eslint/no-var-requires": 1,
+    "@typescript-eslint/no-empty-function": 1,
+    "@typescript-eslint/ban-types": 1,
+    "@typescript-eslint/ban-ts-comment": 1,
```
Comment (yurishkuro) [inline .eslintrc.js:85]: please explain the purpose of these settings. Are they different from some defaults?

I am all for enabling more rues, but it seems this section was previously used to _disabled_ some of them, so I am not clear why we need to enable others, and why specifically these 4 and not more.

File: packages/jaeger-ui/src/components/DeepDependencies/Graph/DdgNodeContent/calc-positioning.tsx [hunk 2]
```diff
@@ -102,7 +102,7 @@ export function _initOpSpan() {
  *     recursive call stack.
  * @returns {number} - Width of narrowest rectangle that contains given words across given lines.
  */
-function calcWidth(lengths: number[], lines: number, longestThusFar: number = 0): number {
+function calcWidth(lengths: number[], lines: number, longestThusFar = 0): number {
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/DeepDependencies/Graph/DdgNodeContent/calc-positioning.tsx:105]: what is the reason for this change? Is a new linter rule complaining about it? It seems like a normal syntax.

Comment (anikdhabal) [reply packages/jaeger-ui/src/components/DeepDependencies/Graph/DdgNodeContent/calc-positioning.tsx:105]: By default it's giving error, called no-inferrable-types. It disallow explicit type declarations for variables or parameters initialized to a number, string, or boolean.

File: .eslintrc.js (hunk 1 above)
Comment (anikdhabal) [reply .eslintrc.js:85]: Because, the new lint is giving error of these 4 types. So, I added these and convert the error to warning. Eslint by default doesn't handle these settings, it's always giving error if we do not handle it.

File: packages/jaeger-ui/src/constants/index.tsx [hunk 3]
```diff
@@ -12,14 +12,14 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-export const TOP_NAV_HEIGHT = 46 as 46;
+export const TOP_NAV_HEIGHT = 46 as const;
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/constants/index.tsx:15]: what's the difference if this simply said `export const TOP_NAV_HEIGHT = 46;`, without `as`?

File: packages/jaeger-ui/src/model/ddg/PathElem.tsx [hunk 4]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-this-alias */
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/model/ddg/PathElem.tsx:1]: No a fan of disabling - can we remove the aliasing instead? And if we must disable, it's should be localized only to where needed, not the whole file.

Comment (anikdhabal) [reply packages/jaeger-ui/src/model/ddg/PathElem.tsx:1]: Ok Understood

File: packages/jaeger-ui/src/constants/index.tsx (hunk 3 above)
Comment (anikdhabal) [reply packages/jaeger-ui/src/constants/index.tsx]: nothing, I just fixed it.

**Review by yurishkuro — DISMISSED** [review]
thanks!

**Comment by yurishkuro** [comment]
looks like linting is failing https://github.com/jaegertracing/jaeger-ui/actions/runs/5836530756/job/15830302191?pr=1565

**Comment by anikdhabal** [comment]
> looks like linting is failing https://github.com/jaegertracing/jaeger-ui/actions/runs/5836530756/job/15830302191?pr=1565

Yes, trying to fix it.

File: packages/jaeger-ui/src/components/DeepDependencies/index.tsx [hunk 5]
```diff
@@ -173,7 +173,7 @@ export class DeepDependencyGraphPageImpl extends React.PureComponent<TProps, TSt
     const { graphState } = this.props;
     const { visEncoding } = this.props.urlState;
 
-    if (graphState && graphState.state === fetchedState.DONE) {
+    if (graphState && graphState.state === fetchedState.DONE && 'model' in graphState) {
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/components/DeepDependencies/index.tsx:176]: this doesn't feel like the right solution to me. Is the reason for linting errors due to missing or incomplete type declaration? Based on the existing code it looks like the `model` field is always expected under these conditions, so why do we need to test for it at runtime, instead of enforce it via type checking?

Comment (anikdhabal) [reply packages/jaeger-ui/src/components/DeepDependencies/index.tsx:176]: Yes,this is the reason for linting errors. Yeah,the is approach in not intuitive. Let me fix it.

Comment (anikdhabal) [reply packages/jaeger-ui/src/components/DeepDependencies/index.tsx:176]: The problem is [TS2339: Property does not exist on type {}]

Comment (yurishkuro) [reply packages/jaeger-ui/src/components/DeepDependencies/index.tsx:176]: why can't we add it to the type declaration?

Comment (anikdhabal) [reply packages/jaeger-ui/src/components/DeepDependencies/index.tsx:176]: Yes, I've added it to the type declaration, now it's fine. Let me fix the PR.

**Comment by anikdhabal** [comment]
@yurishkuro fixed

**Review by yurishkuro — APPROVED** [review]

**Comment by anshgoyalevil** [comment]
286 eslint warnings were introduced with PR
https://github.com/jaegertracing/jaeger-ui/actions/runs/5907286634/job/16024953128#step:6:636
Was this the expected behavior?

**Comment by yurishkuro** [comment]
> 286 eslint warnings were introduced with PR ... Was this the expected behavior?

Not ideal. Perhaps we can fine tune the settings, on case by case basis.
